actions/docker-build/action.yaml

```diff
@@ -0,0 +1,44 @@
+name: "Publish Docker image"
+description: "Use a Dockerfile to build an `image.tar` file and upload it to GitHub"
+
+inputs:
+  dockerfile-dir:
+    description: "Directory containing the Dockerfile"
+    required: true
+  github-token:
+    description: "GitHub token"
+    required: true
+  image-artifact-name:
+    description: "Name of the artifact that contains the Docker image.tar file to push, see https://github.com/actions/upload-artifact (Default is 'image-artifact')."
+    required: false
+    default: "image-artifact"
+  image-name:
+    description: "Name of Docker image (Default is the repository name)."
+    required: false
+    default: ${{ github.event.repository.name }}
+  ref:
+    description: "Ref to checkout"
+    required: false
+    default: ${{ github.ref_name }}
+runs:
+  using: "composite"
+  steps:
+    - name: Check out repository
+      uses: actions/checkout@v3
+      with:
+        ref: ${{ inputs.ref }}
+        token: ${{ inputs.github-token }}
+
+    - name: Build Dockerfile and store them in "build" directory
+      run: |
+        docker build -t ${{ inputs.image-name }}:${{ github.run_id }} ${{ inputs.dockerfile-dir }}/
```

Comment:
Did you test this with setting `dockerfile-dir` to something else than `.`? I'm pretty sure from previous testing that you also need a `--file` flag additionally for other paths to work